SotW bottom banner improvements

mikachan opened the pull request:
This PR adds a few tweaks to enhance the bottom banner:

- Refactor the podcast link spacing so the 'Listen to all episodes' link sits to the left at small resolutions
- Increases the line height of the bottom banner in line with design in Figma
- Adds spacing below headings in line with Figma
- Increases font size of headings at mobile in line with Figma

| Before | After |
| ------- | -------- |
| ![image](https://user-images.githubusercontent.com/1645628/145079961-ab5b86f7-2015-4240-bbba-4986e0ad0533.png) | ![image](https://user-images.githubusercontent.com/1645628/145079817-a6245b54-6be1-4160-9d04-408b7ec647f8.png) |

beafialho commented:
The "After" screenshot looks good to me 👍

ryelle commented:
Can you change the headings back to `h3`? they were changed to `h5`s in this PR #105, but it breaks the heading semantics. I mentioned in [that review that](https://github.com/WordPress/wporg-news-2021/pull/105#pullrequestreview-824574598)

> There should probably be another (visually hidden) h2 for the footer section, and these should be h3.

mikachan commented:
> Can you change the headings back to h3?

I've changed the headings to h3s, and I've added a hidden h2 with the text, "Get the Latest Updates".

I've made the headings `26px` at mobile, but I left the desktop size at the default for h3s, which is `36px`. Is this OK, or should I reduce it back to `26px`? Here's what the larger size looks like:

![image](https://user-images.githubusercontent.com/1645628/145097610-359de3b0-995c-433a-898c-328134aa0b75.png)

ryelle commented on source/wp-content/themes/wporg-news-2021/block-template-parts/footer-archive.html:
```diff
@@ -1,72 +1,80 @@
-		<!-- /wp:heading -->
+<!-- wp:group {"align":"full","classname":"footer-archive"} -->
+<div class="wp-block-group alignfull footer-archive">
+	<!-- wp:heading {"level":2} -->
+	<h2>Get the Latest Updates</h2>
+	<!-- /wp:heading -->
```
```suggestion
	<!-- wp:heading {"level":2,"classname":"screen-reader-text"} -->
	<h2 class="screen-reader-text">Get the Latest Updates</h2>
	<!-- /wp:heading -->
```

You can use the default screen-reader-text class here to hide this

ryelle commented on source/wp-content/themes/wporg-news-2021/sass/components/footer-archive/_footer-archive.scss:
```diff
@@ -1,32 +1,47 @@
-			}
+.footer-archive {
+	h2 {
+		display: none;
```
`display: none` actually hides this content from screen readers too. You need to do [a bit more to visually hide content](https://github.com/WordPress/gutenberg/blob/3475693ae69d3464defa842a76da8505bc275a46/packages/block-library/src/common.scss#L62), which is why the `screen-reader-text` class helper exists 🙂

ryelle replied:
This `display:none` still needs to be removed, it's hiding the heading:

<img width="301" alt="Screen Shot 2021-12-08 at 10 36 53 AM" src="https://user-images.githubusercontent.com/541093/145236887-6f44da63-a856-49e0-9131-1bc0cf2a67c5.png">

ryelle reviewed (changes requested):
> I've made the headings 26px at mobile, but I left the desktop size at the default for h3s, which is 36px.

I think it's supposed to be 26px in both places, according to the design.

mikachan commented:
Brill thanks, I didn't realise the `screen-reader-text` class existed.

I've added a specific size for these headings just for the footer, so they're always `26px`. Ready for another review, thanks for your help!

danieldudzic commented:
Changes look fine. I just have one question: There seems to be a discrepancy (in Figma) in the spacing between the heading and paragraph in the bottom banner:
<img width="1180" alt="px" src="https://user-images.githubusercontent.com/905781/145204371-a14c4a7f-4ebb-4138-9617-2df2f74de923.png">

Not sure which one is correct and should be used.

Requesting guidance from @beafialho :)

beafialho commented:
Sure, we can make them all `10px`

ryelle reviewed (changes requested):
Looking good visually, just [one more request](https://github.com/WordPress/wporg-news-2021/pull/107#discussion_r764979856) 🙂

mikachan commented:
Gah sorry, removed this now! Thanks @ryelle.

ryelle commented:
Merging this, if we want to adjust the margins, it can happen in another PR.

mikachan commented:
The margins mentioned above are already `10px`, so I think everything discussed here has been actioned.
